clang/lib/DPCT/DPCT.cpp

```diff
@@ -542,6 +543,156 @@ void parseFormatStyle() {
   DpctGlobalInfo::setCodeFormatStyle(Style);
 }
 
+static std::string getCmakeBuildPathFromInRoot(StringRef InRoot,
```

Comment:
could you put all cmake script migration logic into a new cpp/hpp files?

Reply:
Okay